cumulusci/core/config/BaseConfig.py

```diff
@@ -2,7 +2,12 @@
 
 
 class BaseConfig(object):
-    """ BaseConfig provides a common interface for nested access for all Config objects in CCI. """
+    """ BaseConfig provides a common interface for nested access for all Config objects in CCI.
+
+    As of 2020 there is a set() method for setting values. Please use that
+    instead of config.config["foo"] = "bar". Directly manipulating the
+    internal config dict will eventually be phased out.
```

Comment:
I'm not convinced yet:
1. It invents a new non-standard Python protocol for setting values, so it's one more thing consumers of BaseConfig need to get familiar with. Also it looks too much like the `set()` builtin for constructing a set object.
2. This docstring says to use set(), but the docstring for set() mentions a situation where you can't use set().
3. I don't see updates elsewhere to actually use this method.
4. We were brainstorming possible larger changes to config objects such as trying to use pydantic or switching to item-style access. I'd like to have consensus on our end goal before starting to make smaller changes.

I think we should keep this change out of this PR and talk about it separately.